src/Persistence_matrix/include/gudhi/Persistence_matrix/columns/set_column.h

```diff
@@ -0,0 +1,1026 @@
+      if (cellTarget->get_element() == Field_operators::get_additive_identity()) {
+        if constexpr (Master_matrix::isNonBasic && !Master_matrix::Option_list::is_of_boundary_type) {
+          if (cellTarget->get_row_index() == chain_opt::get_pivot()) pivotIsZeroed = true;
+        }
+        _delete_cell(itTarget);
+      } else {
+        if constexpr (Master_matrix::Option_list::has_row_access) ra_opt::update_cell(**itTarget);
+        ++itTarget;
+      }
+      ++itSource;
+    }
+  }
+
+  while (itTarget != column_.end()) {
+    operators_->multiply_inplace((*itTarget)->get_element(), val);
+    if constexpr (Master_matrix::Option_list::has_row_access) ra_opt::update_cell(**itTarget);
+    itTarget++;
+  }
+
+  while (itSource != column.end()) {
+    _insert_cell(itSource->get_element(), itSource->get_row_index(), column_.end());
+    ++itSource;
+  }
+
+  return pivotIsZeroed;
+}
+
+template <class Master_matrix>
+template <class Cell_range>
+inline bool Set_column<Master_matrix>::_multiply_source_and_add(const Cell_range& column,
+                                                                           const Field_element_type& val) 
+{
+  if (val == 0u) {
+    return false;
+  }
+
+  bool pivotIsZeroed = false;
+
+  for (const Cell& cell : column) {
+    auto it1 = column_.find(const_cast<Cell*>(&cell));
+    if (it1 != column_.end()) {
+      operators_->multiply_and_add_inplace_back(cell.get_element(), val, (*it1)->get_element());
+      if ((*it1)->get_element() == Field_operators::get_additive_identity()) {
+        if constexpr (Master_matrix::isNonBasic && !Master_matrix::Option_list::is_of_boundary_type) {
+          if ((*it1)->get_row_index() == chain_opt::get_pivot()) pivotIsZeroed = true;
+        }
+        _delete_cell(it1);
+      } else {
+        if constexpr (Master_matrix::Option_list::has_row_access) ra_opt::update_cell(**it1);
+      }
+    } else {
+      Cell* newCell = _insert_cell(cell.get_element(), cell.get_row_index(), column_.end());
+      operators_->multiply_inplace(newCell->get_element(), val);
+    }
+  }
+
+  return pivotIsZeroed;
+}
+
+}  // namespace persistence_matrix
+}  // namespace Gudhi
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Hash method for @ref Gudhi::persistence_matrix::Set_column.
+ * 
+ * @tparam Master_matrix Template parameter of @ref Gudhi::persistence_matrix::Set_column.
+ * @tparam Cell_constructor Template parameter of @ref Gudhi::persistence_matrix::Set_column.
+ */
+template <class Master_matrix>
+struct std::hash<Gudhi::persistence_matrix::Set_column<Master_matrix> > 
+{
+  size_t operator()(const Gudhi::persistence_matrix::Set_column<Master_matrix>& column) const {
+    std::size_t seed = 0;
+    for (const auto& cell : column) {
+      seed ^= std::hash<unsigned int>()(cell.get_row_index() * static_cast<unsigned int>(cell.get_element())) +
+              0x9e3779b9 + (seed << 6) + (seed >> 2);
```

Comment:
It is true that the mixing strategy in boost::hash_combine looks a bit more costly (though it probably has better statistical properties to compensate). What I don't like is seeing many copies of `0x9e3779b9 + (seed << 6) + (seed >> 2)`. If the version in boost doesn't work for us, we can write our own hash_combine (under whatever name) in a single place. This not only helps readability, it also makes it easier to change the hash mixing strategy everywhere at once (unless you expect that each column will want a specific strategy?). If one column (unordered_set?) has a specific reason not to use hash_combine, that's fine, it will be visible (and maybe have a comment to explain it).

Reply:
I factorized the hash method such that we can change it easily for every column (except for unordered_set, vector and heap which need their own version).